core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -402,7 +402,12 @@ private[spark] class Executor(
           // Don't forcibly exit unless the exception was inherently fatal, to avoid
           // stopping other tasks unnecessarily.
           if (Utils.isFatalError(t)) {
-            SparkUncaughtExceptionHandler.uncaughtException(t)
+            if (!isLocal) {
+              Thread.getDefaultUncaughtExceptionHandler.
```

Comment:
Ignore my last comment. If something fatal is there anyways the vm will become unusable. Please merge.